commit aaae8e5e9deae04d0d1980f8e70f64db428f6a5f
Author: Richard Hughes <richard@hughsie.com>
Date:   Mon Oct 3 16:31:07 2011 +0100

    power: Ensure the DPMS state is 'on' at startup
    
    This also has the side-effect of disabling the default DPMS timeouts the session may have set.
    
    Resolves https://bugzilla.gnome.org/show_bug.cgi?id=660482

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 9ffe07f..d0fe7c2 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -3265,6 +3265,8 @@ gboolean
 gsd_power_manager_start (GsdPowerManager *manager,
                          GError **error)
 {
+        gboolean ret;
+
         g_debug ("Starting power manager");
         gnome_settings_profile_start (NULL);
 
@@ -3410,6 +3412,13 @@ gsd_power_manager_start (GsdPowerManager *manager,
         if (manager->priv->x11_screen == NULL)
                 return FALSE;
 
+        /* ensure the default dpms timeouts are cleared */
+        ret = gnome_rr_screen_set_dpms_mode (manager->priv->x11_screen,
+                                             GNOME_RR_DPMS_ON,
+                                             error);
+        if (!ret)
+                return FALSE;
+
         /* coldplug the engine */
         engine_coldplug (manager);
         idle_evaluate (manager);
